### PR TITLE
Changes to use Ironic Stein

### PR DESCRIPTION
**Pull request by elfosardo**
This patch contains changes needed to be able to use Ironic Stein.

It has dependencies from other patches in other related repos
and they should all be merged at the same time, following the order below,
to guarantee a correct behavior.

Full patch set:
https://github.com/openshift-metal3/dev-scripts/pull/680
https://github.com/metal3-io/ironic-image/pull/79
https://github.com/metal3-io/ironic-inspector-image/pull/30
https://github.com/metal3-io/ironic-ipa-downloader/pull/3

### CODE REVIEW

**Review by dtantsur — CHANGES_REQUESTED** [review]
You need to delete mdns-related options. The ironic side has to land first though.

**Comment by dtantsur** [comment]
Ah, and `permit_active_introspection`.

**Comment by elfosardo** [comment]
@dtantsur  done both

**Comment by juliakreger** [comment]
LGTM, We will need to wait for the other pull req to merge.

**Review by juliakreger — APPROVED** [review]

File: Dockerfile
Comment (dtantsur) [inline Dockerfile]: This has to be completely removed.

**Review by dtantsur — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (elfosardo) [reply Dockerfile]: done

**Review by dtantsur — APPROVED** [review]

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/921/

**Review by bfournie — APPROVED** [review]
CI fails with "level=error msg="Error: could not inspect: could not inspect node, node is currently 'inspect failed', last error was 'timeout reached while inspecting the node'", which is expected until the ironic-image patch to remove mdns merges - https://github.com/metal3-io/ironic-image/pull/79

**Comment by dtantsur** [comment]
I've merged the ironic change. Does anyone know how to rerun the CI?

**Comment by juliakreger** [comment]
How to rerun, no clue with how this is setup. I figure merge and if it explodes we can revert. :(

File: Dockerfile
Comment (bfournie) [inline Dockerfile:26]: Thinking about this one more - we can leave in extra_hardware and lldp_basic.  Is there any reason these need be removed for Stein?

Comment (dtantsur) [reply Dockerfile:26]: They're not removed completely, it's just the way github shows the diff (see line 25 below)

Comment (elfosardo) [reply Dockerfile:26]: yeah, this view can be confusing
the only change here is the removal of "&& \\" at the end of the line

Comment (bfournie) [reply Dockerfile:26]: That's what I get for reviewing late at night, thanks.

**Pull request merged**
